### ISSUE TITLE
`decollate_batch` return an empty list

### ISSUE THREAD

**Issue by yiheng-wang-nv** [Project-MONAI/MONAI#3370]
`decollate_batch` will return an empty list, if input is a dict, and one of its value is an empty list.
For example:
```
data = {
    "class": ['ToTensord', 'ToTensord'],
    "id": torch.tensor([1, 2]),
    "orig_size": [],
}
```
`decollate_batch(data)` will return `[]`

This case comes from a classification task, where I used `SupervisedTrainer`, and labels are achieved from `ToNumpyd`. The `data` in the above example belongs to `label_transforms` which is in `engine.state.batch`

**Comment by yiheng-wang-nv** [comment]
For classification tasks, labels may only have one dimension, thus in [this line](https://github.com/Project-MONAI/MONAI/blob/9048166225ab272750815af440060785e6854e4a/monai/transforms/inverse.py#L64), `info[TraceKeys.ORIG_SIZE]` will be `()`, which can lead to the issue above mentioned.

For example:
```
a = np.random.rand(2)
print(a.shape[1:])
```
will get `()`

**Comment by wyli** [comment]
#3373 is one way to address this:
```py
>>>decollate_batch({"a": [1, 2], "b": []}, pad_zip=True, fillvalue=None)
[{'a': 1, 'b': None}, {'a': 2, 'b': None}]
>>>decollate_batch({"a": [1, 2], "b": []}, pad_zip=True, fillvalue=0)
[{'a': 1, 'b': 0}, {'a': 2, 'b': 0}]
```
is this good enough?

**Comment by yiheng-wang-nv** [comment]
Thanks @wyli , I think it solves the issue.

In addition, should this place be modified too? For input that only has one dimension.
https://github.com/Project-MONAI/MONAI/blob/9048166225ab272750815af440060785e6854e4a/monai/transforms/inverse.py#L64

**Comment by wyli** [comment]
> In addition, should this place be modified too? For input that only has one dimension.
> 
> https://github.com/Project-MONAI/MONAI/blob/9048166225ab272750815af440060785e6854e4a/monai/transforms/inverse.py#L64

I feel it makes sense to have [] spatial shape for non-spatial data. or we can use `TraceKey.NONE` but I don't see any use cases that require this value.
